### PR TITLE
Convert candidate user-attributes to JSON

### PR DESCRIPTION
**Pull request by balcirakpeter**
- Added method which converts candidate's user-attributes and
  user-core-attributes to JSON object. It will be used in User
  synchronizations.
- Added test to check if the method works correctly.
- Added json dependency to perun-base pom.xml

### CODE REVIEW

File: perun-base/pom.xml [hunk 1]
```diff
@@ -136,6 +136,13 @@
 			<artifactId>javax.mail</artifactId>
 		</dependency>
 
+		<dependency>
+			<groupId>org.json</groupId>
+			<artifactId>json</artifactId>
+			<version>20140107</version>
+			<scope>compile</scope>
```
Comment (zlamalp) [inline perun-base/pom.xml]: ``compile`` is default dependency type, you don't have to specify it.

File: perun-base/pom.xml [hunk 2]
```diff
@@ -136,6 +136,13 @@
 			<artifactId>javax.mail</artifactId>
 		</dependency>
 
+		<dependency>
+			<groupId>org.json</groupId>
+			<artifactId>json</artifactId>
+			<version>20140107</version>
```
Comment (zlamalp) [inline perun-base/pom.xml]: Can you please use the current version ? I found latest to be ``20190722``. Maybe check, if our parent pom ``spring-boot-starter-parent`` doesn't specify it already.

File: perun-base/src/main/java/cz/metacentrum/perun/core/api/Candidate.java [hunk 3]
```diff
@@ -19,6 +20,9 @@
  */
 public class Candidate extends User {
 
+	private String NS_USER_ATTR_CORE = "urn:perun:user:attribute-def:core:";
+	private String NS_USER_ATTR = "urn:perun:user:attribute-def:";
```
Comment (zlamalp) [inline perun-base/src/main/java/cz/metacentrum/perun/core/api/Candidate.java]: Should be also ``static`` and ``final``.

File: perun-base/pom.xml (hunk 1 above)
Comment (balcirakpeter) [reply perun-base/pom.xml]: Removed

File: perun-base/pom.xml (hunk 2 above)
Comment (balcirakpeter) [reply perun-base/pom.xml]: Changed to 20190722.

File: perun-base/src/main/java/cz/metacentrum/perun/core/api/Candidate.java (hunk 3 above)
Comment (balcirakpeter) [reply perun-base/src/main/java/cz/metacentrum/perun/core/api/Candidate.java]: Fixed.

File: perun-base/pom.xml [hunk 4]
```diff
@@ -136,6 +136,12 @@
 			<artifactId>javax.mail</artifactId>
 		</dependency>
 
+		<dependency>
+			<groupId>org.json</groupId>
+			<artifactId>json</artifactId>
+			<version>20190722</version>
```
[the remark this answers is not in the document]
Comment (balcirakpeter) [reply perun-base/pom.xml]: Done.

**Pull request merged**